mining-buddy.lic

```diff
@@ -95,7 +96,11 @@ class MiningBuddy
   def mine_rooms(rooms)
     rooms.each do |room|
       wait_for_script_to_complete('safe-room') if bleeding?
-      check_repair if mine?(room)
+      if mine?(room)
+        check_repair
+        wait_for_script_to_complete('buff', ['mining']) if @settings.waggle_sets['mining']
```

Comment:
💡 Maybe another DRCA helper? Is this pattern common enough that it would be useful?

```ruby
def foo(settings, buff_set_name)
    return unless settings.waggle_sets[buff_set_name]
    wait_for_script_to_complete('buff', [buff_set_name])
end
```

Reply:
it’s SUPER common in the scripts, that would make it easier converting them all

Reply:
wasnt there some reason not to use settings in common files though? i guess not since common-money is using it now

Reply:
I'm game.  We'd want to use this the exact same way for every waggle call so why not

Reply:
I was thinking about this a bit more.

Can we just migrate all of ;buff into common-arcana?

```
  def cast_waggle(settings, waggle_sets_name = 'default')
    return unless settings.waggle_sets[waggle_sets_name]
    
    spells = settings.waggle_sets[waggle_sets_name]
    return unless spells

    if DRStats.barbarian?
      start_barb_abilities(spells, settings)
    elsif DRStats.thief?
      start_khris(spells, settings)
    else
      spells.each_value do |value|
        next unless value['use_auto_mana']
        check_discern(value, settings)
      end

      cast_spells(spells, settings, settings.waggle_force_cambrinth)
    end
  end
```

That's the entirety of the script.  So in-script things would just call cast_waggle, and players could just cast ;buff.

Reply:
edited to remove some redundancy.

Reply:
It's still best to have a standalone `buff` script since some people use it before/after hunting. However, the entirety of the `buff` script could be a call to the new DRCA helper.

Reply:
Yep, it seems to work